packages/teams-js/src/public/dialog.ts

```diff
@@ -246,94 +362,171 @@ export namespace dialog {
-   *
-   * Convert UrlDialogInfo to DialogInfo to send the information to host in {@linkcode open} API.
-   *
-   * @internal
-   * Limited to Microsoft-internal use
-   *
-   * @beta
-   */
-  export function getDialogInfoFromUrlDialogInfo(urlDialogInfo: UrlDialogInfo): DialogInfo {
-    const dialogInfo: DialogInfo = {
-      url: urlDialogInfo.url,
-      height: urlDialogInfo.size ? urlDialogInfo.size.height : DialogDimension.Small,
-      width: urlDialogInfo.size ? urlDialogInfo.size.width : DialogDimension.Small,
-      title: urlDialogInfo.title,
-      fallbackUrl: urlDialogInfo.fallbackUrl,
-    };
-    return dialogInfo;
-  }
+    /**
+     * Namespace for interaction with adaptive card dialogs that need to communicate with the bot framework
+     *
+     * @beta
+     */
+    export namespace bot {
+      /**
+       * Allows an app to open an adaptive card-based dialog module using bot.
+       *
+       * @param botAdaptiveCardDialogInfo - An object containing the parameters of the dialog module including completionBotId {@link BotAdaptiveCardDialogInfo}.
+       * @param submitHandler - Handler that triggers when the dialog has been submitted or closed.
+       *
+       * @returns a function that can be used to send messages to the dialog.
+       *
+       * @beta
+       */
+      export function open(
+        botAdaptiveCardDialogInfo: BotAdaptiveCardDialogInfo,
+        submitHandler?: DialogSubmitHandler,
+      ): void {
+        ensureInitialized(FrameContexts.content, FrameContexts.sidePanel, FrameContexts.meetingStage);
+        if (!isSupported()) {
+          throw errorNotSupportedOnPlatform;
+        }
 
-  /**
-   * @hidden
-   *
-   * Convert BotUrlDialogInfo to DialogInfo to send the information to host in {@linkcode bot.open} API.
-   *
-   * @internal
-   * Limited to Microsoft-internal use
-   *
-   * @beta
-   */
-  export function getDialogInfoFromBotUrlDialogInfo(botUrlDialogInfo: BotUrlDialogInfo): DialogInfo {
-    const dialogInfo: DialogInfo = {
-      url: botUrlDialogInfo.url,
-      height: botUrlDialogInfo.size ? botUrlDialogInfo.size.height : DialogDimension.Small,
-      width: botUrlDialogInfo.size ? botUrlDialogInfo.size.width : DialogDimension.Small,
-      title: botUrlDialogInfo.title,
-      fallbackUrl: botUrlDialogInfo.fallbackUrl,
-      completionBotId: botUrlDialogInfo.completionBotId,
-    };
-    return dialogInfo;
+        const dialogInfo: DialogInfo = getDialogInfoFromBotAdaptiveCardDialogInfo(botAdaptiveCardDialogInfo);
+
+        sendMessageToParent('tasks.startTask', [dialogInfo], (err: string, result: string | object) => {
+          submitHandler?.({ err, result });
+        });
+      }
+
+      /**
+       * Checks if dialog.adaptiveCard.bot capability is supported by the host
+       *
+       * @returns boolean to represent whether dialog.adaptiveCard.bot is supported
+       */
+      export function isSupported(): boolean {
+        return runtime.supports.dialog
```

Comment:
If we do that it'll return an object `{}` which we do not expect. Also, chaining the statements is a better way from my point of view. Please let me know if you don't agree with this.

Reply:
I was pretty sure that this will return a boolean, but I decided to go verify in typescript playground. Here's a [link](https://www.typescriptlang.org/play?#code/JYOwLgpgTgZghgYwgAgJICUCu5gFsUDeAUMslBHACYD2IANgJ7JwAOwAatAM7C0BcyEJlwAjaAG4SZCjXpNgXADIQA5ogYAVCri4B+ASOrU6FEJNLkqtRsi6YWLalDBcBxUhZnWmrFgFEcMAZ9ZAIAX3MPS1kbX1QQLjA4OjoAEWBk6hUQ8MjPKzlkETgoAGFqSggciKl8mKYEZIgQShLqvOkCm0aU9trO+uQEAAs4MD6PAe8h2gA3bjHeBImor0LKDLosnP666cNxtxrJvcL7VsgVyejpxqhKK+Rj1a6mFQhqRWpGsCWdk6mhVwrEez1ONi2Pz+Rw6N0KWxUehh-ThNmBwDoj1RTHwEF+IBU6CMuCxazRzUwSNCYMBaNoeOAAC9FrRSa9BNRfjBgFDaFTciiyUwWHB3vzdrSfA4AEKYMBgVnIgHYoqIADWAGUkgg1Y9wQ1aNzskqTiqEJgoORwABBBx6yXIGCYFIaKCUw7Ujr65BJET8mk0lUsD4sExswbBqC4BQ8Pnh6YsKDUblhk3e8i4TkQUpwfBQODxwpcChQEaFmxcUZQUDGz2C9mJUUQdjACAAd3LTEgufFyqFjuddFd7v+AIdACtqKAIJQtD37QHYf3uzpyuRO8hZsBKtQN22ICItU4mysamEiEQTGAyNhfvgBBhb3gUABeUJSVgcbhLAQARgANFIdgOE4LhuP0GyZCo4EAnclBHIBkxhIhTyATURBOiACC-LQyAKBq9iOM4M4ABQAJQGEYJhwCA759mAFq0SRbo4PgAB0wFES4bGQQiyAAGT8TerEQBxhGgVwPGbFkbFwWRyAAIQvm+2CVNyIAzpI55EAgfLGKJCIkfh4nEZQ5FkeIQA). You are right that my proposal doesn't return a boolean., but the one in that link does and is (in my opinion) a little easier to read than what we do in most other places in the library.

Ultimately, I'm fine if you want to leave the isSupported checks here the same to match the rest of the library, but we shouldn't feel like we can't update functions like this if we feel this is better or more readable.

Reply:
Sure, I think the way you have mentioned is more readable than existing one. I have updated the code as mentioned.